src/router/routes/errorHandling.ts

```diff
@@ -10,25 +17,39 @@ export const applicationErrorMapper = (
     next: express.NextFunction,
 ) => {
     console.error(err);
-    if (err instanceof ResourceNotFoundError) {
+    if (isResourceNotFoundError(err)) {
         next(new createError.NotFound(err.message));
         return;
     }
-    if (err instanceof ResourceVersionNotFoundError) {
+    if (isResourceVersionNotFoundError(err)) {
         next(new createError.NotFound(err.message));
         return;
     }
-    if (err instanceof InvalidResourceError) {
+    if (isInvalidResourceError(err)) {
         next(new createError.BadRequest(`Failed to parse request body as JSON resource. Error was: ${err.message}`));
         return;
     }
+    if (isUnauthorizedError(err)) {
+        next(new createError.Forbidden(err.message));
+        return;
+    }
     next(err);
 };
 
+const statusToOutcome: Record<number, { severity: IssueSeverity; code: IssueCode }> = {
+    400: { severity: 'error', code: 'invalid' },
+    403: { severity: 'error', code: 'security' },
+    404: { severity: 'error', code: 'not-found' },
+    500: { severity: 'error', code: 'exception' },
+};
+
 export const httpErrorHandler = (err: any, req: express.Request, res: express.Response, next: express.NextFunction) => {
     if (createError.isHttpError(err)) {
         console.error('HttpError', err);
-        res.status(err.statusCode).send(OperationsGenerator.generateError(err.message));
+        const { severity, code } = statusToOutcome[err.statusCode];
+        res.status(err.statusCode).send(
+            OperationsGenerator.generateOperationOutcomeIssue(severity ?? 'error', code ?? 'processing', err.message),
+        );
```

Comment:
L49 will crash if the error code is not in the map. I think that the map covers all the status codes that we are using now but this could break later on. A crash on the error handler is really bad since there's no outer layer to handle it.

I agree with the intent of having a default severity/code of 'error'/'processing', just refactor so that it does not crash.

Reply:
Fixed and tested via [playground](https://www.typescriptlang.org/play?ts=4.0.2#code/MYewdgzgLgBNCGUCuEAqIDySqgLYFMAuGAJX1ACcATAHjCVwCN8KAaGAbznwDcWBLKAE9i0CvzABzANwxQVInCjipMAL4A+GAF5OAKBiGYAFgAMp4lwi8Bw4gHIWFEBXvt5i+xJ7wANvyp7dVYDIzMAZktuPnE7GEcKZ1d3EAUHa2AkWKEgtRCjE1NjKOsYwRF4pxc3OVTPMBAoAFoAMxAkMEDg0MMAVnMSm2yHKuTatMqAD2B8AAcofnBckLVpPT1QSFgnHX0ChGQIAGE6wgAmcz01dc3oTmjbIRSFdV2DlHQsHBACAG0nAB072OdQAujAAPwQ+6lR4jRLVZ6eWbOGYQCASSS5G7gCAgXz4AG+ECSAAUsOySIAlEA)